blog/2023-11-27-magazine-006.md

```diff
@@ -0,0 +1,195 @@
+
+A few use cases within Wing for a singleton include: When adding a provider and ensuring the singular presence of it. Another scenario involves implementing diverse checks for storing resources in a unified global bucket, with the aim of maintaining a singular presence.
+
+Check this out:
+
+```js
+
+ class SingletonBucket {
+   pub static of(scope: std.IResource): cloud.Bucket {
+     let uid = "SingletonBucket";
+     let root = std.Node.of(scope).root;
+     return unsafeCast(root.tryFindChild(uid)) ?? new cloud.Bucket() as uid in root;
+   }
+ }
+
+ let q1 = new cloud.Queue() as "q1";
+ let store1 = SingletonBucket.of(q1);
+
+ test "singleton" {
+   store1.put("hello", "world");
+ }
+```
+
+❤️ [Elad Ben-Israel](https://github.com/eladb) ❤️
+
+
+### Remove `$` from `String` Interpolations
+
+#### This one is a life changer :)
+Before this change, we used the `$` for String interpolations, but not anymore! 
+The motivation behind this one is *one less char to type*, more ergonomic and the `$` char had no practical use.  
+In our context, we're using an expression inside the `string` interpolation so we need a start and end token (the curly braces) but anything else seems redundant.
+
+This is a heck big of a deal! and it probably breaks a lot of Wing apps... but we believe its really worth it!
+
+```js
+
+let s1 = "hello";
+
+// Before
+let s2 = "${s1}, world";
+
+// After
+let s3 = "{s1}, world";
+
+```
+
+❤️ [Yoav Steinberg](https://github.com/yoav-steinberg) ❤️
+
+
+### Expose AWS Resources
+
+Winglang now supports a way to cast a Wing `cloud.Bucket` or any Wing resource into its AWS counterpart. This means that you can actualy grant access to the main AWS resources within Wing Resources.
+
+```js
+
+bring cloud;
+bring aws;
+
+let bucket = new cloud.Bucket();
+if let b = aws.Bucket.from(bucket) {
+  let bucketArn = b.bucketArn;
+}
+
+```
+
+❤️ [Marcio Cruz](https://github.com/marciocadev) ❤️
+
+
+### Sourcemaps and Stacktraces
+
+This change adds sourcemap generation to the compiler. These sourcemaps are used for stacktrace rewriting in both preflight and inflight. 
+Before this change, when testing code using the `expect` keyword, for instance the error message originated from JavaScript, as Wing compiles to JavaScript, and was not very clear and informative.
+So, for the following error:
+
+```js
+bring expect;
+
+let x = 2;
+let y = 3;
+expect.equal(x + y, 8);
+```
+
+You can see below the previous error message. With this change, you can now get a more informative and clear error message:
+![sourcemaps_and_stacktraces_after](./assets/2023-11-27-magazine-006/wing_sourcemaps_stacktraces_after.png)
+
+❤️ [Mark McCulloh](https://github.com/MarkMcCulloh) ❤️
+
+And there's even more excitement waiting for you! Feel free to explore the [The 23rd Wingly Update](https://www.twitch.tv/winglangio) for an extra dose of information. Join in and enjoy!
+
+### Upcoming Events
```

Comment:
Maybe add a section about events from the past two weeks, and say a few words about the react workshop + a link to the recording on youtube